### PR TITLE
Use dyndep API to use non-prerelease WinUI nuget

### PR DESCRIPTION
**Pull request by asklar**

### CODE REVIEW

File: inc/XamlApplication.h [hunk 1]
```diff
@@ -105,4 +107,47 @@ namespace winrt {
...
+      winrt::check_hresult(StringCchPrintf(pfn, std::size(pfn), winuiPackageFamilyNameTemplate.data(), minorVersion));
+
+      winrt::check_hresult(TryCreatePackageDependency(nullptr, pfn, minVersion,
+        PackageDependencyProcessorArchitectures_None, PackageDependencyLifetimeKind_Process,
+        nullptr, CreatePackageDependencyOptions_None, &details::g_winUIDepId));
+    }
+
+    PWSTR packageFullName{};
+
+    winrt::check_hresult(AddPackageDependency(details::g_winUIDepId, 0, AddPackageDependencyOptions_None, &details::g_ctx, &packageFullName));
```
Comment (sylveon) [inline inc/XamlApplication.h:143]: The documentation for AddPackageDependency mentions that `packageFullName` must be manually freed:
> The caller is responsible for freeing this resource once it is no longer needed by calling [HeapFree](https://learn.microsoft.com/en-us/windows/win32/api/heapapi/nf-heapapi-heapfree).

File: inc/XamlApplication.h [hunk 2]
```diff
@@ -105,4 +107,47 @@ namespace winrt {
...
+  /// <param name="minorVersion">Optional minor version. Defaults to 8 (for 2.8)</param>
+  void InitializeWinUI(uint8_t minorVersion = 8) {
+
+    if (details::g_winUIDepId == nullptr) {
+      PACKAGE_VERSION minVersion{};
+      minVersion.Major = minorVersion;
+
+      constexpr const std::wstring_view winuiPackageFamilyNameTemplate{ L"Microsoft.UI.Xaml.2.%d_8wekyb3d8bbwe" };
+      wchar_t pfn[PACKAGE_FAMILY_NAME_MAX_LENGTH]{};
+      winrt::check_hresult(StringCchPrintf(pfn, std::size(pfn), winuiPackageFamilyNameTemplate.data(), minorVersion));
```
Comment (ChrisGuzak) [inline inc/XamlApplication.h:134]: printf expects null terminated string, `wil::zwstring_view` gives you that.
or use `wil::str_printf<std::wstring>(...)` to form the family name. need to include wil/stl.h before wil/resource.h to use the std::wstring form of str_printf.

Comment (sylveon) [reply inc/XamlApplication.h:134]: Or move to C++20 and use `std::format` ;)

Comment (asklar) [reply inc/XamlApplication.h:134]: I'll just use a string literal to get the null-terminated string template 😛 . Ideally everyone would move to cpp20 but... point-in-time

**Comment by sylveon** [comment]
Shouldn't the dyndep APIs be delay linked or dynamically loaded if you want to keep Windows 10 support?

**Comment by asklar** [comment]
@sylveon yeah I think that would make sense. Want to send a PR? :)

**Pull request merged**